petastorm/py_dict_reader_worker.py

```diff
@@ -221,9 +220,14 @@ def _load_rows_with_predicate(self, pq_file, piece, worker_predicate, shuffle_ro
 
             # Merge predicate needed columns with the remaining
             all_cols = [_merge_two_dicts(a, b) for a, b in zip(decoded_other_rows, filtered_decoded_predicate_rows)]
-            return all_cols
+            result = all_cols
         else:
-            return filtered_decoded_predicate_rows
+            result = filtered_decoded_predicate_rows
+
+        if self._transform_spec.func:
+            result = [self._transform_spec.func(row) for row in result]
+
+            return result
```

Comment:
Should this be indented? Are we returning anything here when there isn't a transform function?

Reply:
:+1:  to this

Reply:
Nice catch! The tests were also failing because of that.